src/expressions.md

```diff
@@ -327,96 +429,120 @@ let y = 0...10;
 assert_eq!(x, y);
 ```
 
-## Unary operator expressions
+## Operator expressions
+
+Operators are defined for built in types by the Rust language. Operators will
+panic when they overflow when compiled in debug mode. Many of the following
+operators can be overloaded using traits in `std::ops` or `std::cmp`. Note that
+the traits in `std::cmp` are used more generally to show how a type may be
+compared and will likely be assumed to define actual comparisons by functions
+that use these traits as bounds.
+
+### Unary operator expressions
 
 Rust defines the following unary operators. With the exception of `?`, they are
 all written as prefix operators, before the expression they apply to.
 
 * `-`
   : Negation. Signed integer types and floating-point types support negation. It
     is an error to apply negation to unsigned types; for example, the compiler
-    rejects `-1u32`.
+    rejects `-1u32`. `-` can be overloaded for a type by implementing the
+    `std::ops::Neg` trait.
 * `*`
   : Dereference. When applied to a [pointer](types.html#pointer-types) it
     denotes the pointed-to location. For pointers to mutable locations, the
     resulting [lvalue](expressions.html#lvalues-rvalues-and-temporaries) can be
-    assigned to.  On non-pointer types, it calls the `deref` method of the
-    `std::ops::Deref` trait, or the `deref_mut` method of the
-    `std::ops::DerefMut` trait (if implemented by the type and required for an
-    outer expression that will or could mutate the dereference), and produces
-    the result of dereferencing the `&` or `&mut` borrowed pointer returned
-    from the overload method.
+    assigned to. Dereferencing a raw pointer requires `unsafe`. On non-pointer
+    types `*x` is equivalent to `*std::ops::Deref::deref(&x)` or
+    `*std::ops::Deref::deref_mut(&mut x)` depending on whether the rquired
+    lvalue has to be mutable.
```

Comment:
Should deref coercions be explained here, or in the `Deref` chapter, or elsewhere?